x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/ConfigurationFunction.java

```diff
@@ -7,59 +7,70 @@
 package org.elasticsearch.xpack.sql.expression.function.scalar;
 
 import org.elasticsearch.xpack.sql.expression.Expression;
-import org.elasticsearch.xpack.sql.expression.gen.script.Params;
 import org.elasticsearch.xpack.sql.expression.gen.script.ScriptTemplate;
 import org.elasticsearch.xpack.sql.session.Configuration;
 import org.elasticsearch.xpack.sql.tree.Location;
-import org.elasticsearch.xpack.sql.tree.NodeInfo;
 import org.elasticsearch.xpack.sql.type.DataType;
 import org.elasticsearch.xpack.sql.util.StringUtils;
 
 import java.util.List;
+import java.util.Objects;
 
-abstract class BaseSystemFunction extends ScalarFunction {
+public abstract class ConfigurationFunction extends ScalarFunction {
```

Comment:
Minor: Personally I prefer the `BaseSystemFunction` or plain `SystemFunction`.

Reply:
`Current_timestamp`/`now` are not system functions, yet they share a similar logic.